src/main/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheImpl.java

```diff
@@ -6,20 +6,30 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-class PegoutSignedCacheImpl implements PegoutSignedCache {
+public class PegoutSignedCacheImpl implements PegoutSignedCache {
 
   private static final Logger logger = LoggerFactory.getLogger(PegoutSignedCacheImpl.class);
+  private static final Integer CLEANUP_INTERVAL_IN_HOURS = 1;
 
   private final Map<Keccak256, Instant> cache = new ConcurrentHashMap<>();
+  private final ScheduledExecutorService cleanupScheduler = Executors.newSingleThreadScheduledExecutor();
   private final Duration ttl;
 
-  PegoutSignedCacheImpl(Duration ttl) {
-    assertValidTtl(ttl);
+  public PegoutSignedCacheImpl(Duration ttl) {
+    this.ttl = validateTtl(ttl);
 
-    this.ttl = ttl;
+    // Start a background thread for periodic cleanup
+    cleanupScheduler.scheduleAtFixedRate(
+        this::performCleanup,
+        CLEANUP_INTERVAL_IN_HOURS, // initial delay
+        CLEANUP_INTERVAL_IN_HOURS, // period
+        TimeUnit.HOURS);
```

Comment:
```suggestion
        TimeUnit.HOURS),
);
```

I would also add add trailing comma :). This way next time you add an element there will be a change only in the line you adding.

Reply:
@nathanieliov , that's not valid in Java. It is in Javascript though.

Reply:
Exactly :)

Reply:
It is valid in Java Enums guys :). Try it out.

Reply:
Ohh you both are right. I confused this with a Enum definition :s.